React/Modules/RCTTiming.m

```diff
@@ -233,16 +233,18 @@ - (void)scheduleSleepTimer:(NSDate *)sleepTarget
     _sleepTimer = [[NSTimer alloc] initWithFireDate:sleepTarget
                                            interval:0
                                              target:[_RCTTimingProxy proxyWithTarget:self]
-                                           selector:@selector(timerDidFire)
+                                           selector:@selector(sleepTimerDidFire)
                                            userInfo:nil
                                             repeats:NO];
-    [[NSRunLoop currentRunLoop] addTimer:_sleepTimer forMode:NSDefaultRunLoopMode];
+    dispatch_async(dispatch_get_main_queue(), ^{
+      [[NSRunLoop currentRunLoop] addTimer:_sleepTimer forMode:NSDefaultRunLoopMode];
+    });
   } else {
     _sleepTimer.fireDate = [_sleepTimer.fireDate earlierDate:sleepTarget];
   }
 }
 
-- (void)timerDidFire
+- (void)sleepTimerDidFire
```

Comment:
I'm not 100% sure what to make of this comment — it seems to conflict with your suggestion above to try scheduling on the mainRunLoop (which is owned by the main thread). 

I see that executors don't always run on the same thread, and that my solution here might re-introduce the bug I'm attempting to fix when not running on the WebSocketExecutor, so I'll see if I can come up with something more general.